core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/EntityStatisticsCacheMonitor.scala

```diff
@@ -61,7 +67,8 @@ trait EntityStatisticsCacheMonitor extends LazyLogging {
         logger.info(s"Updated workspace $workspaceId. Cache was ${cacheUpdated - lastModified} millis out of date")
       }
       logger.info(s"Sweep complete. Updated entity cache for ${recordsToUpdate.length} workspace(s), with the limit set to $limit")
-      if(recordsToUpdate.length < limit) HandleBacklog
+      if(recordsToUpdate.isEmpty) ScheduleDelayedSweep
+      else if(recordsToUpdate.length < limit) HandleBacklog
```

Comment:
I believe there's still a logic problem here. If the number of workspaces needing a cache update is an even multiple of `limit`, then this code will toggle between `Sweep` and `ScheduleDelayedSweep`, but never reach `HandleBacklog`, at least until a user triggers some workspace to need a cache update and the counts change.

Some thoughts on simplifying the business logic:
* make the `entity_cache_last_updated` db column non-null, with a default value of 1, and ask liquibase to set all current workspaces to 1 when it runs (for safety, it could set all workspaces to 1 where entity_cache_last_updated != last_modified)
* remove the `limit` configuration value and process workspaces one at a time - you'll always either get 1 or 0.